ethers-contract/tests/contract.rs

```diff
@@ -172,6 +175,100 @@ mod eth_tests {
+        let client3 = connect(&ganache, 2);
+        let client4 = connect(&ganache, 3);
+
+        // create a factory which will be used to deploy instances of the contract
+        let multicall_factory =
+            ContractFactory::new(multicall_abi, multicall_bytecode, client.clone());
+        let simple_factory = ContractFactory::new(abi.clone(), bytecode.clone(), client2.clone());
+        let not_so_simple_factory =
+            ContractFactory::new(not_so_simple_abi, not_so_simple_bytecode, client3.clone());
+
+        // `send` consumes the deployer so it must be cloned for later re-use
+        // (practically it's not expected that you'll need to deploy multiple instances of
+        // the _same_ deployer, so it's fine to clone here from a dev UX vs perf tradeoff)
+        let multicall_deployer = multicall_factory.deploy(()).unwrap();
+        let multicall_contract = multicall_deployer.clone().send().await.unwrap();
+        let simple_deployer = simple_factory.deploy("the first one".to_string()).unwrap();
+        let simple_contract = simple_deployer.clone().send().await.unwrap();
+        let not_so_simple_deployer = not_so_simple_factory
+            .deploy("the second one".to_string())
+            .unwrap();
+        let not_so_simple_contract = not_so_simple_deployer.clone().send().await.unwrap();
+
+        // need to declare the method first, and only then send it
+        // this is because it internally clones an Arc which would otherwise
+        // get immediately dropped
+        simple_contract
+            .connect(client2.clone())
+            .method::<_, H256>("setValue", "reset first".to_owned())
+            .unwrap()
+            .send()
+            .await
+            .unwrap();
+        not_so_simple_contract
+            .connect(client3.clone())
+            .method::<_, H256>("setValue", "reset second".to_owned())
+            .unwrap()
+            .send()
+            .await
+            .unwrap();
+
+        // get the calls for `value` and `last_sender` for both SimpleStorage contracts
+        let value = simple_contract.method::<_, String>("getValue", ()).unwrap();
+        let value2 = not_so_simple_contract
+            .method::<_, (String, Address)>("getValues", ())
+            .unwrap();
+        let last_sender = simple_contract
+            .method::<_, Address>("lastSender", ())
+            .unwrap();
+        let last_sender2 = not_so_simple_contract
+            .method::<_, Address>("lastSender", ())
+            .unwrap();
+
+        // initiate the Multicall instance
+        let multicall = Multicall::new(Some(multicall_contract.address()), None, client4.clone());
+
+        // add calls to multicall
+        let multicall = multicall.add_call(value);
+        let multicall = multicall.add_calls(vec![value2]);
+        let multicall = multicall.add_calls(vec![last_sender, last_sender2]);
+
+        let return_data: (U256, (String, (String, Address), Address, Address)) =
```

Comment:
I think we can drop the block_number return argument, it doesn't feel that useful, and I'm a bit confused by why they added it in the original multicall contract